src/walletbackend/WalletSynchronizer.cpp

```diff
@@ -165,19 +163,19 @@ void WalletSynchronizer::mainLoop()
 
 
         /* if endScanHeight is set, stop syncing at endScanHeight and start syncing from top of the chain */
-        if (endScanHeight != std::nullopt && getCurrentScanHeight() >= endScanHeight)
+        if (m_endScanHeight != std::nullopt && getCurrentScanHeight() >= m_endScanHeight)
```

Comment:
Should be able to simplify this to just

```suggestion
        if (m_endScanHeight && getCurrentScanHeight() >= m_endScanHeight)
```